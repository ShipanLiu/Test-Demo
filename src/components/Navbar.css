.navbar {
  position: sticky;
  top: 0;
  z-index: 999;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 80px;
  font-size: 1.2rem;
  background: linear-gradient(90deg, rgb(28, 27, 27) 0%, rgb(26, 23, 23) 100%);
}

.navbar-container {
  max-width: 1500px;
  height: 80px;
}

.navbar-logo {
  display: flex;
  align-items: center;
  justify-self: start;
  margin-left: 20px;
  color: #fff;
  font-size: 2rem;
  text-decoration: none;
  cursor: pointer;
}

.fa-typo3 {
  margin-left: 0.5rem;
  font-size: 1.8rem;
}

.nav-menu {
  display: flex;
  justify-content: center;
  width: 30vw;
  margin-right: -1rem;
  text-align: center;
  list-style: none;
}

.nav-item {
  height: 80px;
  margin-right: 2rem;
}

.nav-links {
  display: flex;
  align-items: center;
  height: 100%;
  padding: 0.5rem;
  color: #fff;
  font-size: 2rem;
  text-decoration: none;
}

.nav-home {
  margin-right: 0.5rem;
}

.nav-links:hover {
  border-bottom: 4px solid #fff;
  transition: all 0.2s ease-out;
}

.search-bar {
  flex: 1;
  width: 50%;
  height: 2.5rem;
  /* margin-left: 4rem; */
}
.search-bar input {
  width: 80%;
  height: 100%;
}
.search-bar button {
  height: 100%;
}

.fa-cat {
  margin-left: 0.3rem;
  color: #fff;
}

.fa-bars {
  color: #fff;
}

.nav-links-mobile {
  display: none;
}

.menu-icon {
  display: none;
}

/* .navbar-logo {
  animation-name: color-change;
  animation-duration: 5s;
} */

/* @keyframes color-change {
  from {background-color: red;}
  to {background-color: yellow;}
} */

/* 大于960之后， 就是另外一种排版。 */
@media screen and (max-width: 580px) {
  .NavbarItems {
    position: relative;
  }

  .nav-menu {
    position: absolute;
    top: 80px;
    left: -100%;
    display: flex;
    flex-direction: column;
    width: 100%;
    height: 90vh;
    opacity: 1;
    transition: all 0.5s ease;
  }

  .nav-menu.active {
    left: 0;
    z-index: 1;
    background: #242222;
    opacity: 1;
    transition: all 0.5s ease;
  }

  .nav-links {
    display: table;
    width: 100%;
    padding: 1rem;
    text-align: center;
  }

  .nav-links:hover {
    color: #242424;
    background-color: #fff;
    border-radius: 0;
  }

  .navbar-logo {
    position: absolute;
    top: 0;
    left: 0;
    transform: translate(25%, 50%);
  }

  .menu-icon {
    position: absolute;
    top: 0;
    right: 0;
    display: block;
    font-size: 1.8rem;
    transform: translate(-100%, 60%);
    cursor: pointer;
  }

  .fa-times {
    color: #fff;
    font-size: 2rem;
  }

  .nav-links-mobile {
    display: block;
    width: 80%;
    margin: 2rem auto;
    padding: 14px 20px;
    color: #fff;
    font-size: 1.5rem;
    text-align: center;
    text-decoration: none;
    background-color: transparent;
    border: 1px solid #fff;
    border-radius: 4px;
    transition: all 0.3s ease-out;
  }

  .nav-links-mobile:hover {
    color: #242424;
    background: #fff;
    transition: 250ms;
  }
}
