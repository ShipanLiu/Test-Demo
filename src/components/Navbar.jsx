import React, { useState, useEffect } from 'react'
import './Navbar.css'
import { Link, withRouter } from 'react-router-dom'
import {
  HomeOutlined,
  BackwardOutlined,
  ForwardOutlined,
  CloseOutlined,
  UnorderedListOutlined,
} from '@ant-design/icons'
import { Input, Space } from 'antd'
const { Search } = Input

function Navbar(props) {
  const [click, setClick] = useState(false)
  const [button, setButton] = useState(true)
  const [loading, setloading] = useState(false)
  const [searchValue, setSearchValue] = useState('')

  const handleClick = () => setClick(!click)
  const closeMobileMenu = () => setClick(false)

  const showButton = () => {
    if (window.innerWidth <= 960) {
      setButton(false)
    } else {
      setButton(true)
    }
  }

  const onChange = (e) => {
    setSearchValue(e.target.value)
  }

  const onSubmit = (e) => {
    e.preventDefault()
    props.onSearch(searchValue)
    setSearchValue('')
    props.history.push('/products')
    console.log(props)
  }

  const goBackToHome = () => {
    closeMobileMenu()
    return 'All'
  }

  return (
    <>
      <nav className="navbar">
        <Link to="/" className="nav-links nav-home" onClick={goBackToHome}>
          <HomeOutlined />
        </Link>
        <div className="navbar-container">
          <div className="menu-icon" onClick={handleClick}>
            <i className={click ? 'fas fa-times' : 'fas fa-bars'} />
          </div>
          <ul className={click ? 'nav-menu active' : 'nav-menu'}>
            <li></li>
            <li className="nav-item">
              <Link to="/" className="nav-links">
                <BackwardOutlined />
              </Link>
            </li>
            <li className="nav-item">
              <Link to="/" className="nav-links">
                <ForwardOutlined />
              </Link>
            </li>
            <li className="nav-item">
              <Link to="/" className="nav-links" onClick={closeMobileMenu}>
                <CloseOutlined />
              </Link>
            </li>
          </ul>
        </div>
        {/* <Search
          className="search-bar"
          placeholder="input search text"
          // allowClear
          enterButton="Search"
          size="large"
          style={{ marginRight: '5rem' }}
          // loading={loading}
          onSearch={onSearch}
          value={searchValue}
        /> */}
        <form className="search-bar" onSubmit={onSubmit}>
          <input
            type="text"
            onChange={onChange}
            placeholder="search here"
            value={searchValue}
          />
          <button>search</button>
        </form>
      </nav>
    </>
  )
}

export default withRouter(Navbar)
