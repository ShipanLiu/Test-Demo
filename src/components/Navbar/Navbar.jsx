import React, { useState, useRef, useCallback } from 'react'
import { useHistory } from 'react-router-dom'
import './Navbar.css'
// import { Link } from 'react-router-dom'
import {
  HomeOutlined,
  BackwardOutlined,
  ForwardOutlined,
  CloseOutlined,
} from '@ant-design/icons'

function Navbar(props) {
  const [click, setClick] = useState(false)
  let history = useHistory()
  const handleClick = () => setClick(!click)
  const closeMobileMenu = () => setClick(false)
  const inputRef = useRef()

  const onSubmit = useCallback((e) => {
    e.preventDefault()
    const inputValue = inputRef.current.value.trim()
    console.log(inputValue)
    if (inputValue.length === 0) {
      return
    } else {
      inputRef.current.value = ''
      props.onSearch(inputValue)
      history.push('/products')
    }
  })

  const goBackToHome = useCallback(() => {
    closeMobileMenu()
    // // refresh the page
    inputRef.current.value = ''
    history.push('/home')
  })
  const deleteInput = useCallback(() => {
    inputRef.current.value = ''
    history.push('/home')
  })
  return (
    <>
      <nav className="navbar" data-testid="navbar">
        <div className="nav-links nav-home" onClick={goBackToHome}>
          <CloseOutlined />
        </div>
        <div className="navbar-container">
          <div className="menu-icon" onClick={handleClick}>
            <i className={click ? 'fas fa-times' : 'fas fa-bars'} />
          </div>
          <ul className={click ? 'nav-menu active' : 'nav-menu'}>
            <li></li>
            <li className="nav-item">
              <div className="nav-links">
                <BackwardOutlined />
              </div>
            </li>
            <li className="nav-item">
              <div className="nav-links">
                <ForwardOutlined />
              </div>
            </li>
            <li className="nav-item">
              <div to="/" className="nav-links" onClick={deleteInput}>
                <HomeOutlined />
              </div>
            </li>
          </ul>
        </div>
        <form className="search-bar" onSubmit={onSubmit}>
          <input type="text" placeholder="search here" ref={inputRef} />
          <button>search</button>
        </form>
      </nav>
    </>
  )
}

export default Navbar
