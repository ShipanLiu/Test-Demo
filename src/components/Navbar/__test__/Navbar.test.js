import React from 'react'
import ReactDOM from 'react-dom'
import Navbar from '../Navbar'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import renderer from 'react-test-renderer'

afterEach(cleanup)

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Navbar></Navbar>, div)
})

// test wheather the 'onSearch' property was delivered
it('renders perfectly', () => {
  const { getByTestId } = render(<Navbar onSearch></Navbar>)
  expect(getByTestId('navbar')).toHaveProperty
})

it('matches snapshot', () => {
  const tree = renderer.create(<Navbar onSearch></Navbar>).toJSON()
  console.log(tree)
  // create a Snapshot file
  expect(tree).toMatchSnapshot()
})
