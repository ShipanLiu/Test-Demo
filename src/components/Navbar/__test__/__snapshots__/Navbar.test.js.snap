// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="navbar"
  data-testid="navbar"
>
  <div
    className="nav-links nav-home"
    onClick={[Function]}
  >
    <span
      aria-label="close"
      className="anticon anticon-close"
      role="img"
    >
      <svg
        aria-hidden="true"
        data-icon="close"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
        />
      </svg>
    </span>
  </div>
  <div
    className="navbar-container"
  >
    <div
      className="menu-icon"
      onClick={[Function]}
    >
      <i
        className="fas fa-bars"
      />
    </div>
    <ul
      className="nav-menu"
    >
      <li />
      <li
        className="nav-item"
      >
        <div
          className="nav-links"
        >
          <span
            aria-label="backward"
            className="anticon anticon-backward"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="backward"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M485.6 249.9L198.2 498c-8.3 7.1-8.3 20.8 0 27.9l287.4 248.2c10.7 9.2 26.4.9 26.4-14V263.8c0-14.8-15.7-23.2-26.4-13.9zm320 0L518.2 498a18.6 18.6 0 00-6.2 14c0 5.2 2.1 10.4 6.2 14l287.4 248.2c10.7 9.2 26.4.9 26.4-14V263.8c0-14.8-15.7-23.2-26.4-13.9z"
              />
            </svg>
          </span>
        </div>
      </li>
      <li
        className="nav-item"
      >
        <div
          className="nav-links"
        >
          <span
            aria-label="forward"
            className="anticon anticon-forward"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="forward"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M825.8 498L538.4 249.9c-10.7-9.2-26.4-.9-26.4 14v496.3c0 14.9 15.7 23.2 26.4 14L825.8 526c8.3-7.2 8.3-20.8 0-28zm-320 0L218.4 249.9c-10.7-9.2-26.4-.9-26.4 14v496.3c0 14.9 15.7 23.2 26.4 14L505.8 526c4.1-3.6 6.2-8.8 6.2-14 0-5.2-2.1-10.4-6.2-14z"
              />
            </svg>
          </span>
        </div>
      </li>
      <li
        className="nav-item"
      >
        <div
          className="nav-links"
          onClick={[Function]}
          to="/"
        >
          <span
            aria-label="home"
            className="anticon anticon-home"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="home"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
              />
            </svg>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <form
    className="search-bar"
    onSubmit={[Function]}
  >
    <input
      placeholder="search here"
      type="text"
    />
    <button>
      search
    </button>
  </form>
</nav>
`;
